framework/gemini_client.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2025 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+from google import genai
+
+import settings
+
+
+def get_gemini_response(prompt: str, client) -> str:
```

Comment:
Nit: type hint for client.

Reply:
Thoughts on creating your own GeminiClient class? And that class could maintain the client instance and `get_gemini_response` could be an instance method (def get_gemini_response(self, prompt:str))

Then you could also handle the destruction of this client in the `__del__` destructor method (see comment below about closing the client)